rfc/mediachain-rfc-4.md

```diff
@@ -0,0 +1,448 @@
+
+### Moderation and Publishing Model
+
+Namespaces allow us fine grained control on publishing and
+participation level. At one end of the spectrum, we want to
+encourage public namespaces which allow lightweight permissionless
+participation. At the other end, we want to support curated namespaces
+to combat spam and allow authoritative sources and commercial media
+providers to maintain control of their own data.
+
+In order to implement this model, each participating peer has an
+associated identity corresponding to a public key. The identities
+are intended to be long-lived and eventually have an attached reputation
+score.
+
+All statements published in the system are signed with the key of the
+source of the statement. In order to publish in a moderated namespace,
+the originating peer must present a certificate which allows it to
+publish there. Alternatively, it can convince a peer with the right
+certificate to publish on its behalf. Certificates can be obtained by
+moderators and owners of the namespace. 
+
+This moderation scheme, with delegated administrative control,
+attaches a cost to peer identities because certificates can be
+revoked. This encourages cooperative behavior by individual peers,
+thus avoiding the social cost of cheap identities [1]. At the same
+time, individual peers with publishing rights are free to implement
+their own authentication for their clients.
+
+## Namespace-based Queries and Aggregation
+
+TK
+
+
+## A Heterogeneous Network of Cooperative Peers
+
+The distributed database is maintained by a heterogeneous peer-to-peer
+network. Peers are operated by different organizations and
+individuals, and contribute different resources to the network.
+
+Each peer maintains some parts of the dataset and has a limited view
+of the mediachain. Peers synchronize their state with peer-to-peer
+interactions, and converge their view of the data structure by
+exchanging published statements.
+
+Statements can be propagated in the network via push, either with
+direct messages or through streaming. For stream propagation, stable
+peers form pubsub overlays for their namespaces and emit published
+statements to their subscribers. Equally well, peers can poll their
+peers for recent updates, allowing them to operate in a disconnected
+fashion.
+
+The metadata associated with statements is stored in IPLD, with the
+seeding initially supplied by sources. As statements propagate in the
+network, peers may opportunistically reseed metadata in order to aid
+availability and distribution. More specialized peers can provide
+persistent seeding and achiving services for some namespaces, perhaps
+even for a fee.
+
+### Peer Roles
```

Comment:
Thoughts on this section from discussion with @jessewalden: it's quite long, but doesn't provide a ton of information on the system/protocol itself and actually raises more questions of motivation, division of responsibility, etc. Let's split it off into its own document with detailed and concrete examples, and focus on closer to the wire aspects of the protocol here: base roles can be reduced to read, write, aggregate and a separate dimension of rate/regularity.